### DISCUSSION TITLE
Date Range

### DISCUSSION THREAD

**Discussion by kradkovich** [icloud-photos-downloader/icloud_photos_downloader#928] [Q&A]
How do I download using the date range feature? I cannot seem to get it working. Here is my current snippit:

docker run -it --rm --name icloudpd -v $(pwd)/Photos:/data -e TZ=America/Los_Angeles icloudpd/icloudpd:latest icloudpd --directory  /data --username x@x.com  --password x --folder-structure none --skip-live-photos  --watch-with-interval 3600

I was trying to download directly to my external drive but i always get an error: data is read only. So my plan is to download my date ranges as I have over 300gb of photos which exceeds my macs internal storage capacity.

**Comment by AndreyNikiforov** [comment]
date ranges are not currently supported. There are two ways to limit the number of downloads: `--until-found` and `--recent`. Unfortunately neither of them solves your challenge with limited local storage.

I suggest digging into permission issue with external storage - `icloudpd` does nothing special with files or use/need any special/untrivial permission setup. You may get better troubleshooting experience with macos binary as it will remove any docker-related complexity (with permissions), if any.

**Reply by MarkCanada** [reply]
I was looking at this program and it seemed very promising, but the missing date ranges doesn't work too well for my use case.
We have two iPhones and an Andoid in the family and at the end of the month (or whenever I have time) I take the photos of everyone from the past month and compile them into a locally hosted family library. Right now I use USB, but was looking for a better way to transfer. Especially if/when optimized storage will be enabled.
Are downloads by date range planned at some point, or is there a technical problem with the iCloud connection that makes it hard/impossible? 
I probably can work around by leaving some photos in the download location and use "--until-found X". Just thought I'd ask first what the prospects of date ranges appearing are.
